js/modules/require_impl.go

```diff
@@ -0,0 +1,68 @@
+package modules
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/loader"
+)
+
+// OldRequireImpl is an implementation of require that is the old k6 behaviours
+// It happens to not be commonJS compatible as it loads relative to the currentl required file,
+// instead of relative to the file the `require()` is written in.
+type OldRequireImpl struct {
```

Comment:
"Legacy" is a bit more descriptive than "old". Plus `NewOldRequireImpl()` sounds wrong :wink:

```suggestion
// LegacyRequireImpl is a legacy implementation of `require()` that is not compatible with
// CommonJS as it loads modules relative to the currently required file,
// instead of relative to the file the `require()` is written in.
// See https://github.com/grafana/k6/issues/2674
type LegacyRequireImpl struct {
```

Reply:
fixed in 5396f22bb